runtime/src/multisig.rs

```diff
@@ -0,0 +1,415 @@
+        ///
+        /// # Arguments
+        /// * `multisig` - MultiSig address.
+        /// * `signer` - Signer to remove.
+        pub fn remove_multisig_signer(origin, signer: Signer) -> Result {
+            let sender = ensure_signed(origin)?;
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), "Multi sig does not exist");
+            <MultiSigSigners<T>>::insert((sender.clone(), signer), false);
+            Self::deposit_event(RawEvent::MultiSigSignerRemoved(sender, signer));
+            Ok(())
+        }
+
+        /// Change number of sigs required by a multisig. This must be called by the multisig itself.
+        ///
+        /// # Arguments
+        /// * `multisig` - MultiSig address.
+        /// * `sigs_required` - New number os sigs required.
+        pub fn change_sigs_required(origin, sigs_required: u64) -> Result {
+            let sender = ensure_signed(origin)?;
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), "Multi sig does not exist");
+            <MultiSigSignsRequired<T>>::insert(&sender, &sigs_required);
+            Self::deposit_event(RawEvent::MultiSigSignaturesRequiredChanged(sender, sigs_required));
+            Ok(())
+        }
+    }
+}
+
+decl_event!(
+    pub enum Event<T>
+    where
+        AccountId = <T as system::Trait>::AccountId,
+    {
+        /// Event for multi sig creation. (MultiSig address, Creator address, Signers(pending approval), Sigs required)
+        MultiSigCreated(AccountId, AccountId, Vec<Signer>, u64),
+        /// Event for adding a proposal (MultiSig, proposalid)
+        ProposalAdded(AccountId, u64),
+        /// Emitted when a proposal is executed. (MultiSig, proposalid, result)
+        ProposalExecuted(AccountId, u64, bool),
+        /// Signer added (Authorization accepted) (MultiSig, signer_added)
+        MultiSigSignerAdded(AccountId, Signer),
+        /// Multi Sig Signer Authorized to be added (MultiSig, signer_authorized)
+        MultiSigSignerAuthorized(AccountId, Signer),
+        /// Multi Sig Signer removed (MultiSig, signer_removed)
+        MultiSigSignerRemoved(AccountId, Signer),
+        /// Change in signatures required by a multisig (MultiSig, new_sigs_required)
+        MultiSigSignaturesRequiredChanged(AccountId, u64),
+    }
+);
+
+impl<T: Trait> Module<T> {
+    /// Creates a new proposal
+    pub fn create_proposal(
+        multisig: T::AccountId,
+        proposal: Box<T::Proposal>,
+        sender_signer: Signer,
+    ) -> Result {
+        ensure!(
+            Self::ms_signers((multisig.clone(), sender_signer)),
+            "not an signer"
+        );
+        let proposal_id = Self::ms_tx_done(multisig.clone());
+        <Proposals<T>>::insert((multisig.clone(), proposal_id), proposal);
+        // Since proposal_ids are always only incremented by 1, they can not overflow.
+        let next_proposal_id: u64 = proposal_id + 1u64;
```

Comment:
Same remark about `overflowing_add` here. Would not be time-consuming to take care of.

Reply:
It's much easier to just implement it and move on than to debate but this is something that I don't think we should be doing. Please see my above comment. Perhaps we can have a quick call to discuss this further.

Reply:
It's OK to keep `+` as long as we don't overflow in debug mode.